prometheus/promsafe/promauto_adapter/promauto_adapter.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2024 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package promauto_adapter provides compatibility adapter for migration of calls of promauto into promsafe
+package promauto_adapter
```

Comment:
I agree, the `promauto_adapter` idea is very weak. 
But the issue does exist - my intention was to support both APIs: the `prometheus` and `promauto`. So if you had either `prometheus.NewCounterVec` or `promauto.NewCounterVec` - you can easily switch to `promsafe.NewCounterVec` (using `named import` in case of promauto_adapter)